### PR TITLE
Update the colours to the latest spec.

### PR DESCRIPTION
**Pull request by bwinton**
And generate better Android file.
And generate Swift file.

🙂

### CODE REVIEW

**Comment by bwinton** [comment]
If this seems decent to you, I'll add the rest of the alpha colours, and bump the version number…

**Comment by bwinton** [comment]
(Uh, it should be noted that this used the old values for the blue-50, so _really_ shouldn't be merged until that gets updated.)

**Comment by bwinton** [comment]
Well, the colours have been updated to the new ones and I've added all the alpha versions from the old file, so it's time for a @michaelmccombie review-ping… 😉

File: tokens/_aliases.yml [hunk 1]
```diff
@@ -2,107 +2,140 @@ aliases:
...
-  color-yellow-90: '#960e18'
-  
-  color-green-5: '#e3fff3'
-  color-green-10: '#d1ffee'
+  color-green-05: '#e3fff3'
+  color-green-10: '#d1ffed'
   color-green-20: '#b3ffe3'
   color-green-30: '#88ffd1'
-  color-green-40: '#54ffbd'
+  color-green-40: '#53ffbd'
```
Comment (shorlander) [inline tokens/_aliases.yml]: #54FFBD is the right Green 40

File: tokens/_aliases.yml [hunk 2]
```diff
@@ -2,107 +2,140 @@ aliases:
...
-  color-green-90: '#08403f'
-  
-  color-blue-5: '#aaf2ff'
+  color-green-90: '#083f35'
+
+  color-blue-05: '#aaf2ff'
   color-blue-10: '#80ebff'
   color-blue-20: '#00ddff'
-  color-blue-30: '#00b3f4'
+  color-blue-30: '#00b3f3'
```
Comment (shorlander) [inline tokens/_aliases.yml]: #00B3F4 is the right Blue 30

File: tokens/_aliases.yml [hunk 3]
```diff
@@ -2,107 +2,140 @@ aliases:
...
   color-blue-40: '#0090ed'
   color-blue-50: '#0060df'
-  color-blue-60: '#0250bb'
-  color-blue-70: '#054096'
-  color-blue-80: '#073072'
+  color-blue-60: '#0150bb'
```
Comment (shorlander) [inline tokens/_aliases.yml]: Blue 60 = #0250BB

File: tokens/_aliases.yml [hunk 4]
```diff
@@ -2,107 +2,140 @@ aliases:
...
+  color-blue-70: '#053f96'
```
Comment (shorlander) [inline tokens/_aliases.yml]: Blue 70 = #054096

File: tokens/_aliases.yml [hunk 5]
```diff
@@ -2,107 +2,140 @@ aliases:
...
+  color-blue-80: '#063071'
```
Comment (shorlander) [inline tokens/_aliases.yml]: Blue 80 = #073072

File: tokens/_aliases.yml [hunk 6]
```diff
@@ -2,107 +2,140 @@ aliases:
...
   color-violet-30: '#c689ff'
   color-violet-40: '#ab71ff'
-  color-violet-50: '#9059ff'
-  color-violet-60: '#7542e5'
-  color-violet-70: '#592acb'
-  color-violet-80: '#45278d'
-  color-violet-90: '#321c64'
-  
-  color-purple-5: '#f7e2ff'
+  color-violet-50: '#9058ff'
```
Comment (shorlander) [inline tokens/_aliases.yml]: Violet 50 = #9059FF

File: tokens/_aliases.yml [hunk 7]
```diff
@@ -2,107 +2,140 @@ aliases:
...
+  color-violet-60: '#7541e5'
```
Comment (shorlander) [inline tokens/_aliases.yml]: Violet 60 = #7542E5

File: tokens/_aliases.yml [hunk 8]
```diff
@@ -2,107 +2,140 @@ aliases:
...
+  color-violet-70: '#582acb'
```
Comment (shorlander) [inline tokens/_aliases.yml]: Violet 70 = #592ACB

File: tokens/_aliases.yml [hunk 9]
```diff
@@ -2,107 +2,140 @@ aliases:
...
+  color-violet-80: '#44268d'
```
Comment (shorlander) [inline tokens/_aliases.yml]: Violet 80 = #45278D

File: tokens/_aliases.yml [hunk 10]
```diff
@@ -2,107 +2,140 @@ aliases:
...
+  color-violet-90: '#311b63'
```
Comment (shorlander) [inline tokens/_aliases.yml]: Violet 90 = #321C64

File: tokens/_aliases.yml [hunk 11]
```diff
@@ -2,107 +2,140 @@ aliases:
...
   color-purple-10: '#f6b8ff'
   color-purple-20: '#f68fff'
-  color-purple-30: '#f770ff'
-  color-purple-40: '#d74cf0'
+  color-purple-30: '#f565ff'
+  color-purple-40: '#d74cef'
   color-purple-50: '#b833e1'
   color-purple-60: '#952bb9'
-  color-purple-70: '#722291'
+  color-purple-70: '#712191'
```
Comment (shorlander) [inline tokens/_aliases.yml]: Purple 70 = #722291

File: tokens/_aliases.yml [hunk 12]
```diff
@@ -2,107 +2,140 @@ aliases:
...
-  color-gray-10: '#f9f9fa'
-  color-gray-20: '#ededf0'
-  color-gray-30: '#cdcdd4'
-  color-gray-40: '#b1b1bc'
-  color-gray-50: '#9595a3'
-  color-gray-60: '#7a7a8b'
-  color-gray-70: '#5e5e72'
-  color-gray-80: '#42425a'
-  color-gray-90: '#202340'
+  color-pink-05: '#ffdeef'
```
Comment (shorlander) [inline tokens/_aliases.yml]: Pink 05 = #FFDEF0

File: tokens/_aliases.yml [hunk 13]
```diff
@@ -2,107 +2,140 @@ aliases:
...
+  color-pink-10: '#ffb3db'
```
Comment (shorlander) [inline tokens/_aliases.yml]: Pink 10 = #FFB4DB

File: tokens/_aliases.yml [hunk 14]
```diff
@@ -2,107 +2,140 @@ aliases:
...
+  color-pink-20: '#ff8ac5'
+  color-pink-30: '#ff6bba'
+  color-pink-40: '#ff49a2'
```
Comment (shorlander) [inline tokens/_aliases.yml]: Pink 40 = #FF4AA2

File: tokens/_aliases.yml [hunk 15]
```diff
@@ -2,107 +2,140 @@ aliases:
...
+  color-pink-50: '#ff298a'
+  color-pink-60: '#e31587'
+  color-pink-70: '#c60083'
```
Comment (shorlander) [inline tokens/_aliases.yml]: Pink 70 = #C60084

File: tokens/_aliases.yml [hunk 16]
```diff
@@ -2,107 +2,140 @@ aliases:
...
+  color-pink-80: '#7f145b'
+  color-pink-90: '#50134a'
```
Comment (shorlander) [inline tokens/_aliases.yml]: Pink 90 = #50134B

**Comment by bwinton** [comment]
So, it seems like there's a bunch of stuff wrong in the conversion… Can you show me in the sketch file where "Violet 50" is defined?

Screen Shot 2019-04-19 at 11 13 58

File: tokens/_aliases.yml [hunk 17]
```diff
@@ -2,107 +2,140 @@ aliases:
...
+
+  color-red-05: '#ffdfe6'
```
Comment (shorlander) [inline tokens/_aliases.yml]: Red 05 = #FFDFE7

File: tokens/_aliases.yml [hunk 18]
```diff
@@ -2,107 +2,140 @@ aliases:
...
+  color-red-10: '#ffbdc5'
+  color-red-20: '#ff9aa2'
+  color-red-30: '#ff838b'
+  color-red-40: '#ff6975'
+  color-red-50: '#ff4f5e'
```
Comment (shorlander) [inline tokens/_aliases.yml]: Red 50 = #FC4055

File: tokens/_aliases.yml [hunk 19]
```diff
@@ -2,107 +2,140 @@ aliases:
...
+  color-red-60: '#e22850'
+  color-red-70: '#c50041'
```
Comment (shorlander) [inline tokens/_aliases.yml]: Red 70 = #C50042

File: tokens/_aliases.yml [hunk 20]
```diff
@@ -2,107 +2,140 @@ aliases:
...
+  color-red-80: '#810120'
+  color-red-90: '#440205'
```
Comment (shorlander) [inline tokens/_aliases.yml]: Red 90 = #440306

File: tokens/_aliases.yml [hunk 21]
```diff
@@ -2,107 +2,140 @@ aliases:
...
+
+  color-orange-05: '#fff3de'
```
Comment (shorlander) [inline tokens/_aliases.yml]: Orange 05 = #FFEFDE

File: tokens/_aliases.yml [hunk 22]
```diff
@@ -2,107 +2,140 @@ aliases:
...
+  color-orange-10: '#ffd5b2'
```
Comment (shorlander) [inline tokens/_aliases.yml]: Orange 10 = #FFDDBD

File: tokens/_aliases.yml [hunk 23]
```diff
@@ -2,107 +2,140 @@ aliases:
...
+  color-orange-20: '#ffb587'
```
Comment (shorlander) [inline tokens/_aliases.yml]: Orange 20 = #FFC999

File: tokens/_aliases.yml [hunk 24]
```diff
@@ -2,107 +2,140 @@ aliases:
...
+  color-orange-30: '#ffa266'
```
Comment (shorlander) [inline tokens/_aliases.yml]: Orange 30 = #FFB675

File: tokens/_aliases.yml [hunk 25]
```diff
@@ -2,107 +2,140 @@ aliases:
...
+  color-orange-40: '#ff8a50'
```
Comment (shorlander) [inline tokens/_aliases.yml]: Orange 40 = #FF9640

File: tokens/_aliases.yml [hunk 26]
```diff
@@ -2,107 +2,140 @@ aliases:
...
+  color-orange-50: '#ff7139'
```
Comment (shorlander) [inline tokens/_aliases.yml]: Orange 50 = #FF7B39

File: tokens/_aliases.yml [hunk 27]
```diff
@@ -2,107 +2,140 @@ aliases:
...
+  color-orange-60: '#e25820'
```
Comment (shorlander) [inline tokens/_aliases.yml]: Orange 60 = #E25920

**Comment by bwinton** [comment]
Are they in a library somewhere?
Screen Shot 2019-04-19 at 11 17 05

File: tokens/_aliases.yml [hunk 28]
```diff
@@ -2,107 +2,140 @@ aliases:
...
+  color-orange-70: '#cc3c00'
```
Comment (shorlander) [inline tokens/_aliases.yml]: Orange 70 = #CC3D00

File: tokens/_aliases.yml [hunk 29]
```diff
@@ -2,107 +2,140 @@ aliases:
...
+  color-orange-80: '#9e280a'
```
Comment (shorlander) [inline tokens/_aliases.yml]: Orange 80 = #9E280B

File: tokens/_aliases.yml [hunk 30]
```diff
@@ -2,107 +2,140 @@ aliases:
...
+  color-orange-90: '#7c1504'
+
+  color-yellow-05: '#ffffcc'
+  color-yellow-10: '#ffff98'
+  color-yellow-20: '#ffea80'
```
Comment (shorlander) [inline tokens/_aliases.yml]: Yellow 20 = #FFF36E

File: tokens/_aliases.yml [hunk 31]
```diff
@@ -2,107 +2,140 @@ aliases:
...
+  color-yellow-30: '#ffd567'
```
Comment (shorlander) [inline tokens/_aliases.yml]: Yellow 30 = #FFDE67

File: tokens/_aliases.yml [hunk 32]
```diff
@@ -2,107 +2,140 @@ aliases:
...
+  color-yellow-40: '#ffbd4f'
+  color-yellow-50: '#ffa335'
```
Comment (shorlander) [inline tokens/_aliases.yml]: Yellow 50 = #FFA436

File: tokens/_aliases.yml [hunk 33]
```diff
@@ -2,107 +2,140 @@ aliases:
...
+  color-yellow-60: '#e27f2e'
+  color-yellow-70: '#c35926'
```
Comment (shorlander) [inline tokens/_aliases.yml]: Yellow 70 = #C45A27

File: tokens/_aliases.yml [hunk 34]
```diff
@@ -2,107 +2,140 @@ aliases:
...
+  color-yellow-80: '#a7341f'
+  color-yellow-90: '#960d17'
```
Comment (shorlander) [inline tokens/_aliases.yml]: Yellow 90 = #960E18

**Comment by craigcook** [comment]
Sorry for neglecting this PR for so long. Protocol has been on the back burner for a while but I'm trying to bring it to the front again and crank up the heat.

Looking at https://mozilla.design/firefox/color/ I see lots of colors with slightly different hex values from what's here. Then there's also the Sketch file in Abstract and it has a some different values as well. 

I believe the Sketch file is the canonical origin, but ultimately we should strive to make these tokens the single source of truth for these colors going forward. Ideally the mozilla.design site would consume the tokens directly but that's not how it was built (don't get me started). So I propose we look through this PR one more time and match everything to the Sketch brand guide, then merge and publish. Then I can work on updating mozilla.design to match the tokens.

**Comment by bwinton** [comment]
So, I generated the colours from the Sketch file, which means they _should_ be in sync (and are shorlander-approved!)… If there's a new Sketch file, please throw it my way, and I'll re-generate them, and post a new PR.

**Comment by craigcook** [comment]
> So, I generated the colours from the Sketch file, which means they _should_ be in sync (and are shorlander-approved!)… If there's a new Sketch file, please throw it my way, and I'll re-generate them, and post a new PR.

The one I'm looking at is https://app.goabstract.com/projects/750f6600-36fd-11e9-8a3e-7ff09cd48be1/branches/master/files/5E9E3A26-D8B0-4929-8463-F6723EFE1E46 and specifically I'm looking at page 4.2 and selecting squares to get the hex value, and almost every single one is just a digit or two off from the values in this PR. 

image

But I don't see you in the user list in Abstract so you may not have access to the link above. I'll send you the file directly but perhaps you also need access to Abstract for versioning. This PR is pretty old and maybe everything has changed since then. 

And now I also wonder if there are multiple brand guide Sketch files making the rounds, and if so I don't know which one of *those* is canonical. And even if we're looking at the same thing after all, depending on how you're generating this from Sketch it might be possible the sample squares on this page are different from the colors defined somewhere else in the same file, in which case that sounds like a whole other problem.

So anyway, yeah, I just want to figure out where these colors are coming from and ultimately make one token library the single source of truth for everything. If the values in this PR are correct right now, then this Sketch file and mozilla.design are currently wrong.

**Comment by bwinton** [comment]
Okay, I went through the latest sketch file, and updated all the colours to match the ones in the accessibility sections (since those have hex values beside them), so this should be good to merge, in theory. 😉

**Pull request merged**